src/Common/src/Common.Expression/Spring/Ast/OpDec.cs

```diff
@@ -0,0 +1,157 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the Apache 2.0 License.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Steeltoe.Common.Expression.Internal.Spring.Ast
+{
+    public class OpDec : Operator
+    {
+        private readonly bool _postfix;  // false means prefix
+
+        public OpDec(int startPos, int endPos, bool postfix, params SpelNode[] operands)
+            : base("--", startPos, endPos, operands)
+        {
+            _postfix = postfix;
+            if (operands == null || operands.Length == 0)
+            {
+                throw new InvalidOperationException("Operands can not be empty");
+            }
+        }
+
+        public override ITypedValue GetValueInternal(ExpressionState state)
+        {
+            var operand = LeftOperand;
+
+            // The operand is going to be read and then assigned to, we don't want to evaluate it twice.
+            var lvalue = operand.GetValueRef(state);
+
+            var operandTypedValue = lvalue.GetValue();  // operand.getValueInternal(state);
+            var operandValue = operandTypedValue.Value;
+            var returnValue = operandTypedValue;
+            ITypedValue newValue = null;
+
+            if (IsNumber(operandValue))
+            {
+                if (operandValue is decimal)
```

Comment:
I think a [pattern matching switch statement](https://docs.microsoft.com/en-us/dotnet/csharp/pattern-matching#using-pattern-matching-switch-statements) would improve readability here

Reply:
I'll update on next PR that finishes this and cleans up commented out code ... etc.